lib/src/where_type.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+/// Emits only the events which have type [R].
+///
+/// If the source stream is a broadcast stream the result will be as well.
+///
+/// Errors from the source stream are forwarded directly to the result stream.
+StreamTransformer<Null, R> whereType<R>() => _WhereType<R>();
+
+class _WhereType<R> extends StreamTransformerBase<Null, R> {
```

Comment:
Oh I see you commented about this, I don't fully understand the issue here but I can check out the repo and try it out which would probably make it more obvious.

Reply:
See some discussion here:
https://github.com/dart-lang/language/issues/180#issuecomment-461998427

concrete scenario:

```dart
  var numbers = Stream<num>.fromIterable([1, 1.5, 2]);
  var ints = numbers.transform(whereType<int>());
```

In this case the `transform` excepts an argument assignable to `StreamTransformer<num, S>`. Since the `whereType` call doesn't have a way to statically know the `num` there, it either needs us to be explicit about it (`numbers.transform(whereType<num, int>()`), or to use some default that can fill in for any `T` that may have been on the stream. `Null` is the only possibility there for now, I'm not sure if we've figured out what the user referenceable  bottom type will be after NNBD...

Reply:
I would at least leave a length comment here, its going to be pretty bizarre for anybody reading this in the future so some context would certainly help :).

Reply:
Added a note to the doc comment, how does that look to you? I think for further details the `git blame` should point here which will have extra detail.

Reply:
I think what you added is fine